pkg/cmd/miscellaneous.go

```diff
@@ -139,6 +139,23 @@ func getShootClusterName() (clustername string) {
 	return clustername
 }
 
+// getProjectNamespace returns the project namespace
+func getProjectNamespace() (shootNamespace string) {
+	apiAdress := ""
+	file, _ := os.Open(getKubeConfigOfCurrentTarget())
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), "server:") {
+			apiAdress = strings.TrimPrefix(scanner.Text(), "server: ")
+		}
+	}
+	pieces := strings.Split(apiAdress, ".")
+
+	return "garden-" + pieces[2]
```

Comment:
I do not know other option how to take the project namespace from the garden cluster

Reply:
If you know the project name, you can retrieve the namespace out of its spec, e.g. .spec.namespace

Reply:
hmm, what about when the user is targeting shoot directly.
e.g. `g target garden -> seed -> shoot`. In this way I do not know the project name

Reply:
The URL of the API is not a better solution, it will even not work when the DNS is disabled and the public address of the `kube-apiserver` service is used directly.

Reply:
The `kubeconfig` file contains the technical name of the shoot as cluster and context name, e.g. `shoot-[-]<project-name-[-]<shoot-name>`. You have the shoot name in the current target, so you can easily find the project name from the technical name.